shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/network/target/UserTargetType.kt

```diff
@@ -0,0 +1,36 @@
+package co.nimblehq.blisskmmic.data.network.target
+
+import co.nimblehq.blisskmmic.BuildKonfig
+import co.nimblehq.blisskmmic.data.network.helpers.TargetType
+import io.ktor.http.*
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+sealed class UserTargetType: TargetType
+
+class LoginTargetType(email: String, password: String): UserTargetType() {
+
+    @Serializable
+    data class LoginInput(
+        @SerialName("grant_type")
+        val grantType: String,
+        val email: String,
+        val password: String,
+        @SerialName("client_id")
+        val clientId: String,
+        @SerialName("client_secret")
+        val clientSecret: String
+    )
+
+    override var baseURL: String = BuildKonfig.BASE_URL
+    override var path: String = "oauth/token"
+    override var method: HttpMethod = HttpMethod.Post
+    override var body: Any? = LoginInput(
+        "password",
+        email,
+        password,
+        BuildKonfig.CLIENT_ID,
+        BuildKonfig.CLIENT_SECRET
+    )
+    override var headers: Map<String, String>? = null
```

Comment:
hence we need the updates

```suggestion
    override val baseURL: String = BuildKonfig.BASE_URL
    override val path: String = "oauth/token"
    override val method: HttpMethod = HttpMethod.Post
    override val body: Any = LoginInput(
        "password",
        email,
        password,
        BuildKonfig.CLIENT_ID,
        BuildKonfig.CLIENT_SECRET
    )
    override val headers: Map<String, String>? = null
```